### DISCUSSION TITLE
Hosting Teslamate for Free on GCP VM and Securing with Cloudflare Access

### DISCUSSION THREAD

**Discussion by danielraffel** [teslamate-org/teslamate#3612] [Show and tell]
I decided to move Teslamate to an E2 micro-instance (free) server on GCP and configure SSL with CloudFlare Tunnel. To secure Teslamate I setup Cloudflare Access. Figured I'd share if this is useful for anyone as the steps I took to configure and deploy are more or less described here.

https://danielraffel.me/2024/01/12/cloudflare-access/

**Comment by DerRheingold** [comment]
Hi, how is your performance with the E2 micro-instance? I have constant problems of the service crashing when I refresh a page or so. Guess it might be the memory limitations?

**Comment by GtHbbrr** [comment]
Hi DerRheingold, performance on my E2 micro is good and I do not have your issues. I did upgrade my system to postgresql 16 this summer and also upgraded some other items in accordance with the latest instructions for setting up TM on GCP.

Best regards

Op 4 okt 2024 om 10:20 heeft DerRheingold ***@***.***> het volgende geschreven:

﻿

Hi, how is your performance with the E2 micro-instance? I have constant problems of the service crashing when I refresh a page or so. Guess it might be the memory limitations?

**Comment by damianopezzotti** [comment]
Do you use docker?

**Comment by gobeti** [comment]
Great initiative, is there a guide you put together for non-IT people? I´d like to set up teslamate in GCP but I am not even sure which VM to create.... I am afraid I would need some guidelines step by step.. I am sure this could be useful for many people.
